docs/typescript.md

```diff
@@ -158,34 +173,57 @@ const App = () => (
 import React, { SFC } from 'react'
 import styled from 'react-emotion'
 
-type ComponentProps = {
-  className?: string,
-  label: string
+type ComponentProps0 = {
+  className?: string;
+  label: string;
 }
 
-const Component: SFC<ComponentProps> = ({ label, className }) => (
+const Component0: SFC<ComponentProps0> = ({ label, className }) => (
   <div className={className}>{label}</div>
 )
 
 type StyledComponentProps = {
   bgColor: string
 };
 
-const StyledComponent0 = styled(Component)`
+const StyledComponent00 = styled<StyledComponentProps, ComponentProps0>(Component)`
```

Comment:
Why is a prop type passed here but in other places `div` is passed (shouldn't this be a component type)?

Reply:
@mitchellhamilton This is for compatibilities, and, sadly, old typings did so...
https://github.com/emotion-js/emotion/blob/24fa6ed67314ae0b8d79276f7ff2815341f40caa/packages/react-emotion/types/index.d.ts#L101-L108

Reply:
Is this the preferred way to do it now?
If not, could you add a note saying that it's only for compatibility and tell people what they should use now.

Reply:
@mitchellhamilton OK. I will add the note.

Reply:
@mitchellhamilton
I notice that this is the preferred way.... There is no walkaround for this case, to be compatible with old one.
I will try better typing at upcoming emotion@10